### PR TITLE
Fr us 16

### LINKED ISSUE #19: User Story 16, User can log out
As a registered user, merchant, or admin
When I visit the logout path
I am redirected to the welcome / home page of the site
And I see a flash message that indicates I am logged out
Any items I had in my shopping cart are deleted

### PR DESCRIPTION
**Pull request by fredrondina**
Completed US 16 plus tests. Added cart clearing functionality and flash message to session#destroy

closes #19

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -20,6 +20,10 @@ def create
 
   def destroy
     session[:user_id] = nil
+    current_user = nil
+    session[:cart] = {}
+    cart.contents = {} if cart.contents != {}
+    flash[:success] = "Logout Succesful"
```
Comment (brian-greeson) [inline app/controllers/sessions_controller.rb:26]: This makes sense.

**Review by brian-greeson — APPROVED** [review]
This looks good. Lets merge it!

File: app/controllers/sessions_controller.rb (hunk 1 above)
Comment (kelshadarby) [inline app/controllers/sessions_controller.rb:26]: Very nice reset of everything!

File: spec/features/sessions/logout_spec.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        expect(page).to_not have_link("Logout")
+        expect(page).to have_link("Login")
+        expect(current_path).to eq(root_path)
+        expect(page).to have_content("Logout Succesful")
+      end
+
+      after(:all) do
+        UserDetail.destroy_all
+        User.destroy_all
+      end
```
Comment (kelshadarby) [inline spec/features/sessions/logout_spec.rb:46]: This looks great!

**Review by kelshadarby — APPROVED** [review]

**Review by brian-greeson — APPROVED** [review]
Looks good. Lets merge it.

**Pull request merged**